src/hera/shared/_type_util.py

```diff
@@ -92,15 +92,24 @@ def get_unsubscripted_type(t: Any) -> Any:
     return t
 
 
-def origin_type_issubclass(cls: Any, type_: type) -> bool:
-    """Return True if cls can be considered as a subclass of type_."""
-    unwrapped_type = unwrap_annotation(cls)
+def origin_type_issubtype(annotation: Any, type_: Union[type, Tuple[type, ...]]) -> bool:
+    """Return True if annotation is a subtype of type_."""
```

Comment:
should be better to write a note for tuple args.

maybe
`type_ may be tuple of types. In that case, return true if annotation is a subtype of union of items in type_.`